beacon_chain/state/state_transition.py

```diff
@@ -330,26 +348,64 @@ def fill_recent_block_hashes(active_state: ActiveState,
     )
 
 
-def compute_cycle_transitions(
-        crystallized_state: CrystallizedState,
-        active_state: ActiveState,
-        block: 'Block',
-        config: Dict[str, Any]=DEFAULT_CONFIG) -> Tuple[CrystallizedState, ActiveState]:
-    while block.slot_number >= crystallized_state.last_state_recalc + config['cycle_length']:
-        crystallized_state, active_state = initialize_new_cycle(
-            crystallized_state,
-            active_state,
-            block,
-            config=config,
-        )
-        if ready_for_dynasty_transition(crystallized_state, block, config):
-            crystallized_state = compute_dynasty_transition(
-                crystallized_state,
-                block,
-                config
-            )
+def apply_rewards_and_penalties(crystallized_state: CrystallizedState,
+                                active_state: ActiveState,
+                                block: 'Block',
+                                config: Dict[str, Any]=DEFAULT_CONFIG) -> List['ValidatorRecord']:
+    validators = crystallized_state.validators
+    updated_validators = deepcopy(crystallized_state.validators)
+    active_validator_indices = get_active_validator_indices(
+        crystallized_state.current_dynasty,
+        validators
+    )
 
-    return crystallized_state, active_state
+    time_since_finality = block.slot_number - crystallized_state.last_finalized_slot
+    total_deposits = crystallized_state.total_deposits
+    reward_quotient = config['base_reward_quotient'] * int(sqrt(total_deposits))
+    quadratic_penalty_quotient = int(sqrt(config['sqrt_e_drop_time'] / config['slot_duration']))
+
+    last_state_recalc = crystallized_state.last_state_recalc
+    block_vote_cache = active_state.block_vote_cache
+
+    # FFG Rewards
```

Comment:
if we refactor `apply_rewards_and_penalties` to be two functions - `calculate_ffg_rewards` and `calculate_crosslink_rewards`, it probably will be easier to test FFG rewards logic.